pypeit/core/mosaic.py

```diff
@@ -0,0 +1,312 @@
+    else:
+        _tforms = tforms
+
+    msgs.info(f'Constructing image mosaic with {nimg} images and output shape {mosaic_shape}.')
+
+    if ivar is not None:
+        var = [inverse(_ivar) for _ivar in ivar]
+
+    mosaic_npix = np.zeros(mosaic_shape, dtype=int)
+    mosaic_data = np.zeros(mosaic_shape, dtype=float)
+    # NOTE: "mosaic_ivar" is actually the variance until it's inverted just
+    # before output
+    mosaic_ivar = None if ivar is None else np.zeros(mosaic_shape, dtype=float)
+
+    # TODO: These loops can end up creating and destroying lots of big arrays.
+    # Is there a way to make this faster?
+    for i in range(nimg):
+        _inv_tform = np.linalg.inv(_tforms[i])
+        # TODO: I'm not a fan of using cval=np.nan, but this might be the most
+        # robust way to catch mosaic pixels that are not filled by the input
+        # image.  I'm not sure if we lose efficiency by using NaN instead of a
+        # real scalar.
+        _tform_img = ndimage.affine_transform(imgs[i], _inv_tform, output_shape=mosaic_shape,
+                                              cval=np.nan, order=order)
+        filled = np.logical_not(np.isnan(_tform_img))
+        if bpm is not None:
+            _tform_gpm = ndimage.affine_transform(np.logical_not(bpm[i]).astype(float), _inv_tform,
+                                                  output_shape=mosaic_shape, cval=0., order=0)
+            filled &= _tform_gpm > 0.
+        mosaic_data[filled] += _tform_img[filled]
+        mosaic_npix[filled] += 1
+        if ivar is not None:
+            # NOTE: "mosaic_ivar" is actually the variance until it's inverted
+            # just before output
+            mosaic_ivar[filled] += ndimage.affine_transform(var[i], _inv_tform,
+                                                            output_shape=mosaic_shape,
+                                                            cval=0., order=order)[filled]
+
+    # TODO: This test is crude.  Input and output pixel sizes should be
+    # identical, but I don't know if the order=0 approach will ever lead to a
+    # single mosaic pixel being filled by more than one input pixel.  If so,
+    # `overlap='error'` will catch both those cases and when multiple input
+    # images overlap.
+    has_overlap = np.any(mosaic_npix > 1)
+    if has_overlap and overlap == 'error':
+        # Input images should not be allowed to overlap
+        msgs.error('Mosaic has pixels with contributions by more than one input image!')
+
+    filled = mosaic_npix > 0
+    # Average the overlapping pixels
+    if has_overlap:
+        mosaic_data[filled] /= mosaic_npix[filled]
+    if mosaic_ivar is not None:
+        # Propagate the error by averaging the variances
+        if has_overlap:
+            mosaic_ivar[filled] /= mosaic_npix[filled]
+        # Revert to inverse variance
+        mosaic_ivar = inverse(mosaic_ivar)
+
+    return mosaic_data, mosaic_ivar, mosaic_npix, _tforms
```

Comment:
In my opinion you should also be returning a gpm here (or a bpm), where the gpm is 

gpm =  mosaic_npix > 0

Reply:
It makes more sense to me to do this in the calling function.